ogr/ogrsf_frmts/geojson/ogrgeojsonwriter.cpp

```diff
@@ -871,10 +871,41 @@ json_object *OGRGeoJSONWriteAttributes(OGRFeature *poFeature,
             const size_t nLen = strlen(pszStr);
             poObjProp = nullptr;
             if (eSubType == OFSTJSON ||
-                (pszStr[0] == '{' && pszStr[nLen - 1] == '}') ||
-                (pszStr[0] == '[' && pszStr[nLen - 1] == ']'))
+                ((pszStr[0] == '{' && pszStr[nLen - 1] == '}') ||
+                 (pszStr[0] == '[' && pszStr[nLen - 1] == ']')))
             {
-                OGRJSonParse(pszStr, &poObjProp, false);
+                if (poFeature->GetNativeMediaType() &&
+                    strcmp(poFeature->GetNativeMediaType(),
+                           "application/vnd.geo+json") == 0 &&
+                    poFeature->GetNativeData())
+                {
+                    json_object *poNativeObjProp = nullptr;
+                    OGRJSonParse(poFeature->GetNativeData(), &poNativeObjProp,
+                                 false);
```

Comment:
If there would be a lot of fields, doing the above parsing could potentially be costly. It could perhaps be done once in the loop

Reply:
done